rasa_nlu/evaluate.py

```diff
@@ -502,60 +520,91 @@ def run_cv_evaluation(td, n_folds, nlu_config):
     :return: dictionary with key, list structure, where each entry in list
               corresponds to the relevant result for one fold
     """
-    from sklearn import metrics
     from collections import defaultdict
     import tempfile
 
     trainer = Trainer(nlu_config)
     train_results = defaultdict(list)
     test_results = defaultdict(list)
-
+    entity_train_results = defaultdict(lambda: defaultdict(list))
+    entity_test_results = defaultdict(lambda: defaultdict(list))
     tmp_dir = tempfile.mkdtemp()
 
     for train, test in generate_folds(n_folds, td):
-        trainer.train(TrainingData(training_examples=train,
-                                   entity_synonyms=td.entity_synonyms,
-                                   regex_features=td.regex_features))
+        train = TrainingData(training_examples=train,
+                             entity_synonyms=td.entity_synonyms,
+                             regex_features=td.regex_features)
+        test = TrainingData(training_examples=test,
+                            entity_synonyms=td.entity_synonyms,
+                            regex_features=td.regex_features)
+        trainer.train(train)
         model_dir = trainer.persist(tmp_dir)
         interpreter = Interpreter.load(model_dir, nlu_config)
 
         # calculate train accuracy
-        compute_metrics(interpreter, train, train_results)
+        compute_metrics(interpreter, train, train_results, entity_train_results)
         # calculate test accuracy
-        compute_metrics(interpreter, test, test_results)
+        compute_metrics(interpreter, test, test_results, entity_test_results)
 
         utils.remove_model(model_dir)
 
     os.rmdir(os.path.join(tmp_dir, "default"))
     os.rmdir(tmp_dir)
 
-    return CVEvaluationResult(dict(train_results), dict(test_results))
+    return (CVEvaluationResult(dict(train_results), dict(test_results)),
+            CVEvaluationResult(dict(entity_train_results),
+                               dict(entity_test_results)))
 
 
-def compute_metrics(interpreter, corpus, results):
+def compute_metrics(interpreter, corpus, intent_results, entity_results):
```

Comment:
let's try to do this in a more functional way where we do not modify the incoming parameter.

So lets rather return a dict for the intent results, e.g. `return {"Accuracy": ... ,"F1-Score": ...}` and combine it in the calling method.